SmartPlaces.Facilities/samples/Telemetry.Mapped/src/Processors/TelemetryIngestionProcessor.cs

```diff
@@ -318,47 +323,65 @@ private dynamic TranslateType(TypedValue sourceData, DTEntityKind targetType)
                         case DTEntityKind.String:
                             return sourceData.EnumValue.ToString();
                     }
+
                     break;
                 case TypedValue.ValueOneofCase.Float32Value:
                     switch (targetType)
                     {
                         case DTEntityKind.Double:
-                            return Convert.ToDouble(sourceData.Float32Value);
+                            if (sourceData.Float32Value.ToString() != "NaN")
+                            {
+                                return Convert.ToDouble(sourceData.Float32Value);
+                            }
+
+                            logger.LogError("Received NaN as Float32Value");
+                            break;
                         case DTEntityKind.Duration:
                             return TimeSpan.FromHours(sourceData.Float32Value);
                         case DTEntityKind.String:
                             return sourceData.Float32Value.ToString();
                     }
+
                     break;
                 case TypedValue.ValueOneofCase.Float64Value:
                     switch (targetType)
                     {
                         case DTEntityKind.Double:
-                            return sourceData.Float64Value;
+                            if (sourceData.Float32Value.ToString() != "NaN")
```

Comment:
Are we receiving `NaN` because a value is misclassified? I agree we should handle the case because its a possibility, but is this a `Reach out to Mapped Support` sort of exception we should be throwing? Why are we receiving `NaN`?

Reply:
That I don't know. Wanted to add the logging to see what was causing the code issue.